src/System.IO.Pipes/tests/AnonymousPipesThrowsTests.cs

```diff
@@ -0,0 +1,509 @@
+            using (AnonymousPipeServerStream dummyserver = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                Assert.Throws<ArgumentException>(delegate
+                {
+                    AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out, dummyserver.SafePipeHandle, pipeHandle);
+                });
+            }
+        }
+
+        [Fact]
+        public static void ServerPipeDirectionThrows()
+        {
+            using (AnonymousPipeServerStream dummyserver = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                Assert.Throws<NotSupportedException>(delegate
+                {
+                    AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.InOut, dummyserver.SafePipeHandle, null);
+                });
+            }
+        }
+
+        [Fact]
+        public static void ServerReadModeThrows()
+        {
+            // force different constructor path
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream())
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => server.ReadMode = (PipeTransmissionMode)999);
+
+                Assert.Throws<NotSupportedException>(() => server.ReadMode = PipeTransmissionMode.Message);
+            }
+        }
+
+        [Fact]
+        public static void ServerWriteBufferNullThrows()
+        {
+            // force different constructor path
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out, HandleInheritability.None))
+            {
+                Assert.Throws<ArgumentNullException>(() => server.Write(null, 0, 1));
+            }
+        }
+
+        [Fact]
+        public async static void ServerWriteAsyncBufferNullThrows()
+        {
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                await Assert.ThrowsAsync<ArgumentNullException>(async () => await server.WriteAsync(null, 0, 1));
+            }
+        }
+
+        [Fact]
+        public static void ServerWriteNegativeOffsetThrows()
+        {
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => server.Write(new byte[5], -1, 1));
+
+                // array is checked first
+                Assert.Throws<ArgumentNullException>(() => server.Write(null, -1, 1));
+            }
+        }
+
+        [Fact]
+        public async static void ServerWriteAsyncNegativeOffsetThrows()
+        {
+            using (AnonymousPipeServerStream server = new AnonymousPipeServerStream(PipeDirection.Out))
+            {
+                await Assert.ThrowsAsync<ArgumentOutOfRangeException>(async () => await server.WriteAsync(new byte[5], -1, 1));
```

Comment:
The async on the lambda and the await in the delegate aren't actually necessary here... the delegate just needs to return a Task, which WriteAsync does.